chanzuckerberg/idseq-web
# Multi stage pipeline

# Description

Breaking the pipeline into multiple trackable stages so we can know the progress better and utilize the computational resources more efficiently. 
## Type of change

- [ X] Breaking change (fix a feature that knowingly causes existing functionality to not work as expected)

# Test and Reproduce
Outline the steps to test or reproduce the PR here.

1. Upload a sample 
2. look if a new pipeline_run is created sample.pipeline_runs.first 
3. pipeline_run.run_stages should be created and first stage command will be submitted
4. run pipeline_run.update_job_status repeated until pipeline_run.finalized = 1 

## Impacted Areas in Application
List general components of the application that this PR will affect:

Which pages?

- [X ] Sample Details Page
- [X] Pipeline Run debugging pages 

Specific components 

*  Breaking the pipeline to multiple stages. 

# Testing Script:

###Login Page
* Successfully login to the application and get taken to the project page

###Single Upload
* Using an AWS path that doesn't exist throws an error
* Succesfully uploaded files from a folder (use test host)
* Uploading a duplicate file throws an error
* Pipeline gets initiated successfully 

###Batch Upload
* Using an AWS path that doesn't exist throws an error
* Succesfully uploaded files from a folder (use test host)
* Navigate to batch upload settings page after upload 
* Host selection from previous page still selected 
* Uploading a duplicate file throws a warning
* Pipeline gets initiated successfully 

###All Project Page
* Switch between projects 
* Search for a sample
* Navigate through different pages of samples
* Select Upload new sample button
* Select sample to see generated report

###Sample Details Page
* Information is consistent from Project Page
* Notes are editable and save successfully 
* Can navigate to report page 

###Report Page
* Data loads succesfully 
* Collapse and Uncollapse button works
* Clicking on the taxonomy name gives you total reads for that taxon
* Category filter works
* Genus Search works
* Filters work as expected
* Disabling Filters works
* Filtering by column works 

# Checklist:
